src/transformers/modeling_utils.py

```diff
@@ -2955,6 +2955,19 @@ def from_pretrained(
                 **kwargs,
             )
         else:
+            # In case one passes a config to `from_pretrained` + "attn_implementation"
+            # override the `_attn_implementation` attribute to `attn_implementation` of the kwargs
+            # Please see: https://github.com/huggingface/transformers/issues/28038
+
+            # Overwrite `config._attn_implementation` by the one from the kwargs --> in auto-factory
+            # we pop attn_implementation from the kwargs but this handles the case where users
+            # passes manually the config to `from_pretrained`.
+            config = copy.deepcopy(config)
+
+            if kwargs.get("attn_implementation", None) is not None and config._attn_implementation != kwargs.get(
+                "attn_implementation", None
+            ):
+                config._attn_implementation = kwargs.get("attn_implementation", None)
```

Comment:
Why are we getting so many times? 😅 

```suggestion
            kwarg_attn_imp = kwargs.get("attn_implementation", None)
            if kwarg_attn_imp is not None and config._attn_implementation != kwarg_attn_imp:
                config._attn_implementation = kwarg_attn_imp
```